posts/modular-monolith-microservices.md

```diff
@@ -0,0 +1,76 @@
+
+![Monolith Vs Microservice Architecture Diagram](../images/2022/12/modular-monolith.png) ***Modular Monolith Architecture***
+
+## Microservices
+
+No longer considered a new architectural model, microservices have proven to be successful and adopted in approximately [63%](https://dzone.com/articles/new-research-shows-63-percent-of-enterprises-are-a) of enterprises. Most companies who have converted have mentioned that their new architecture has improved customer & end-to-end experience and cut costs on infrastructure and other development tools. 
+
+It is essential to recognize that breaking down a monolith and developing microservices can be expensive and time-consuming. Understanding the benefits of transitioning to microservices and why other organizations are doing the same is vital. Some reasons companies should leverage microservices may include the following: 
+
+- Continuous deployment & continuous integration
+- Testing becomes easier with smaller components to test
+- Ability to switch tech stacks over time
+- Teams can develop & test their components without dependency on other teams
+
+Microservices follow the idea of segregation of duty, isolating data to the service that owns it. Each service should be created for a single responsibility, should be independent of others, & needs the ability to handle service-to-service communication in the event a single service fails.
+
+While proven to be incredibly cost-efficient over time, microservices do not come without their challenges. Maintaining & testing data between services can be difficult, as applications that follow this architecture model can become increasingly complex. Fortunately, some services and applications help companies manage, develop, and deploy microservices for both back and front-end needs.
+
+Modular architecture is no longer new and has proven to succeed for enterprises across the globe. Getting started with maintaining them, however, can be tricky for certain companies that don't know where to begin.  Today we will take a look at services that organizations can utilize for microservice deployment, creation, and management.
+
+## Entando
+
+Entando is an open-source Application Composition Platform that allows users to create, curate, and compose applications for business capabilities. Entando is built on top of Kubernetes which allows developers to not only build their applications but run and deploy them as well. There are various components users can create such as microservices, micro front ends, UI/UX widgets, page templates, and so on. Entando seeks to remove redundant, repetitive code and empower developers to reuse code that can be used across applications.
```

Comment:
The Entando section talks about what is Entando, I would expand on how Entando utilizes or adopts microservices, and is better than or how it solves the problem with the monolithic architecture; basically tying it all together how Entando is the best solution